packages/connector-jsdom/src/connector.ts

```diff
@@ -77,6 +78,25 @@ export default class JSDOMConnector implements IConnector {
         this.server = server;
         this._timeout = server.timeout;
         this._resourceLoader = new CustomResourceLoader(this);
+        this._subprocesses = new Set();
+
+        /*
+         * In case of a SIGTERM, close jsdom and all the subprocesses.
+         * This event doesn't work on windows, but windows automatically
+         * closes the subprocesses.
+         * https://nodejs.org/dist/latest-v10.x/docs/api/process.html#process_process_kill_pid_signal
+         */
+        process.on('SIGTERM', () => {
```

Comment:
I don't think `analyze.ts` is the best place. Services like the online scanner doesn't use that file, it use `engine.ts`. Anyways, let me try something, maybe there is no need on having this in any file.

Reply:
This is what I'm trying, and I think it is a good solution.

The change in this file is that the `close` will call `this.killAllSubprocesses();` as we are doing now, but we don't subscribe to any `process.on(signal)`.

In that way, when we call to engine.close(), it will kill all the subprocess that are still alive.

Then the online scanner or any other service is the one in charge to call to `engine.close()`.

Reply:
👍